src/lib/code-sample/go.ts

```diff
@@ -0,0 +1,167 @@
+
+  if (goPackageName !== DEFAULT_GO_PACKAGE_NAME && isReqWithParams) {
+    const nestedPackageImport = `import ${goPackageName} "${GO_PACKAGE_BASE_PATH}/${goPackageName}"`
+    imports.push(nestedPackageImport)
+  }
+
+  return imports.join('\n')
+}
+
+const getRequestStructName = (path: string): string => {
+  const requestStructNameSuffix = 'Request'
+
+  return isPathNested(path)
+    ? `${pascalCase(removeUntilSecondSlash(path))}${requestStructNameSuffix}`
+    : `${pascalCase(path)}${requestStructNameSuffix}`
+}
+
+const removeUntilSecondSlash = (str: string): string =>
+  str.replace(/^\/[^/]*/, '')
+
+const formatGoArgs = (jsonParams: NonNullJson): string =>
+  Object.entries(jsonParams as Record<string, Json>)
+    .map(([paramKey, paramValue]) => {
+      const formattedValue = formatGoValue({ value: paramValue, key: paramKey })
+      return `${pascalCase(paramKey)}: ${formattedValue}`
+    })
+    .join(', ')
+
+const formatGoValue = ({
+  value,
+  key,
+}: {
+  value: Json
+  key: string
+}): string => {
+  if (value == null) return 'nil'
+  if (typeof value === 'string') return `api.String("${value}")`
+  if (typeof value === 'boolean') return `api.Bool(${value})`
+  if (typeof value === 'number') return `api.Float64(${value})`
+
+  if (Array.isArray(value)) {
+    return formatGoArray(value, key)
+  }
+
+  if (typeof value === 'object') {
+    return formatGoObject(value, key)
+  }
+
+  throw new Error(`Unsupported type: ${typeof value}`)
+}
+
+const formatGoArray = (value: Json[], key: string): string => {
+  if (value.length === 0) {
+    // in Go there's no way define an empty array without specifying type
+    // and code samples definitions don't include the type annotations
+    return 'nil'
+  }
+
+  const formattedItems = value.map((v) => formatGoValue({ value: v, key }))
+  const item = value[0]
+  if (item == null) {
+    throw new Error(`Null value in response array for '${key}'`)
+  }
+
+  const arrayType = isPrimitiveValue(item)
+    ? getPrimitiveTypeName(item)
+    : `api.${pascalCase(key)}`
+
+  return `[${value.length}]${arrayType}{${formattedItems.join(', ')}}`
+}
+
+const isPrimitiveValue = (value: Json): boolean =>
+  value != null && typeof value !== 'object'
+
+const getPrimitiveTypeName = (value: Json): string => {
+  switch (typeof value) {
+    case 'string':
+      return 'string'
+    case 'number':
+      return 'float64'
+    case 'boolean':
+      return 'bool'
+    default:
+      throw new Error(`Unsupported type: ${typeof value}`)
+  }
+}
+
+const formatGoObject = (value: Record<string, Json>, key: string): string => {
+  if (Object.keys(value).length === 0) {
+    return 'struct{}{}'
+  }
+
+  const formattedEntries = Object.entries(value)
+    .map(
+      ([objKey, val]) =>
+        `${pascalCase(objKey)}: ${formatGoValue({ value: val, key: objKey })}`,
+    )
+    .join(', ')
+
+  return `api.${pascalCase(key)}{${formattedEntries}}`
+}
+
+export const createGoResponse = createJsonResponse
```

Comment:
Looks like GO is serializes responses to JSON https://github.com/seamapi/go/blob/e470714a98a413f888a059409a5fd5b9c2383303/core/core.go#L186

Reply:
It still has to put the data into a Go struct, e.g., https://github.com/seamapi/go/blob/e470714a98a413f888a059409a5fd5b9c2383303/connectwebviews/client.go#L81 https://github.com/seamapi/go/blob/e470714a98a413f888a059409a5fd5b9c2383303/connect_webviews.go#L246

See https://drstearns.github.io/tutorials/gojson/

Basically, Go has some syntax to make parsing JSON into structs simpler, but in the GO code, you still interact with the data as a struct, e.g.,

```go
    fmt.Printf(
        "%s",
        Device.ConnectedAccountID,
    )

Reply:
We can merge this with the json response type for now, and update it to `go` in another PR.

Reply:
https://github.com/seamapi/blueprint/issues/97